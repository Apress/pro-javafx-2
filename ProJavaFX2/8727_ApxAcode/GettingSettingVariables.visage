/*
 * @test
 * @run
 */

// GettingSettingVariables.visage
import visage.reflect.*;

public class Point {
  public var x: Number;
  public var y: Number;
}

public var a:String;

public function run() {
  var p = Point { x: 3, y: 4 };

  // Working with instance variable x
  var context = VisageLocal.getContext();
  var classType = context.findClass("GettingSettingVariables.Point");
  var xVar = classType.getVariable("x");
  xVar.setValue(context.mirrorOf(p), context.mirrorOf(7.0));
  println("p.x={p.x}.");

  // Working with script variable a
  var moduleType = context.findClass("GettingSettingVariables");
  var aVar = moduleType.getVariable("a");
  aVar.setValue(context.mirrorOf(p), context.mirrorOf("Hello, World"));
  println("a={a}.");
}
