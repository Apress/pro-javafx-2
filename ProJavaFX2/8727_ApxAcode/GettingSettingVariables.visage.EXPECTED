p.x=7.0.
a=Hello, World.
